src/main/java/com/microsoft/jenkins/function/commands/ZipDeployCommand.java

```diff
@@ -34,7 +34,9 @@ public void execute(IZipDeployCommandData context) {
             tempDir = workspace.createTempDir(ZIP_FOLDER_NAME, null);
             final FilePath zipPath = tempDir.child(ZIP_NAME);
             final DirScanner.Glob globScanner = new DirScanner.Glob(filePattern, excludedFilesAndZip());
-            workspace.zip(zipPath.write(), globScanner);
+            final FilePath sourceDir = workspace.child(Util.fixNull(context.getSourceDirectory()));
```

Comment:
what is sourceDirectory? from user's input? on jenkins agent?

Reply:
It is an input of the user. It should be a relative path to the workspace.